.github/workflows/pull_request.yaml

```diff
@@ -42,10 +42,10 @@ jobs:
       - uses: actions/checkout@v2
       - run: |
           git fetch --prune --unshallow
-      - name: Set up Go 1.16
+      - name: Set up Go 1.18
         uses: actions/setup-go@v1
         with:
-          go-version: 1.16.3
+          go-version: 1.18
```

Comment:
do you think we can move to the changes that jenny added that sets the go version to that which is in the mod file?

Reply:
[This one](https://github.com/solo-io/solo-projects/pull/4640)
Looks like it checks rather than sets, but no reason not to

Reply:
yeah i just added a check that's done when you run codegen. but what i think nathan's referring to is the [issue](https://github.com/solo-io/gloo/issues/7926) i filed for GHA updates (which hasn't been done yet). in there i suggested we use setup-go's `go-version-file`. we could try that here

Reply:
seems like a nice to have, but probably not worth blocking on for a repo that's had one PR in the last ~18 months